DependencyInjection/Configuration.php

```diff
@@ -217,10 +217,11 @@ private function configureDbalDriverNode(ArrayNodeDefinition $node): void
             ->children()
                 ->scalarNode('url')->info('A URL with connection information; any parameter value parsed from this string will override explicitly set parameters')->end()
                 ->scalarNode('dbname')->end()
-                ->scalarNode('host')->defaultValue('localhost')->end()
-                ->scalarNode('port')->defaultNull()->end()
-                ->scalarNode('user')->defaultValue('root')->end()
-                ->scalarNode('password')->defaultNull()->end()
+                ->scalarNode('host')->info('Defaults to "localhost" at runtime.')->end()
+                ->scalarNode('port')->info('Defaults to null at runtime.')->end()
+                ->scalarNode('user')->info('Defaults to "root" at runtime.')->end()
+                ->scalarNode('password')->info('Defaults to null at runtime.')->end()
+                ->booleanNode('override_url')->info('Allows overriding parts of the "url" parameter with dbname, host, port, user, and/or password parameters.')->end()
```

Comment:
```suggestion
                ->booleanNode('override_url')->defaultValue(false)->info('Allows overriding parts of the "url" parameter with dbname, host, port, user, and/or password parameters.')->end()
```

This should default to false and deprecation should be triggered from Configuration, via something like this
```php
->beforeNormalization()
    ->ifTrue(static function ($v) {
        return ! $v['override_url'] && isset($v['url']);
    })
    ->then(static function ($v) {
        Deprecation::trigger(...);

        return $v;
    })
->end();
```

Reply:
done